test/src/test/java/hudson/util/ArgumentListBuilder2Test.java

```diff
@@ -61,4 +70,33 @@ public void slaveMask() throws Exception {
         System.out.println(out);
         assertTrue(out.toString().contains("$ java ********"));
     }
+
+    @Test
+    public void ensureArgumentsArePassedToCommandUnmodified() throws Exception {
+        String[] specials = new String[] {
+                "~", "!", "@", "#", "$", "%", "^", "&", "*", "(", ")",
+                "_", "+", "{", "}", "[", "]", ":", ";", "\"", "'", "\\", "|",
+                "<", ">", ",", ".", "/", "?", " "
+        };
+
+        ArgumentListBuilder args = new ArgumentListBuilder(JavaEnvUtils.getJreExecutable("java"), "-cp", "target/test-classes/", "hudson.util.EchoCommand");
+        args.add(specials);
+        if (Functions.isWindows()) {
```

Comment:
the only thing here is this will give different results on different platforms.

I would preffer 2 tests with an assume(Functions.is...) so that the test when it fails its obvious that it is a platform related issue.

That said - this test is better then no test at all